pkg/controllers/clusterresourceplacement/clusterresourceplacement_controller.go

```diff
@@ -211,6 +368,19 @@ func parsePolicyIndexFromLabel(s *fleetv1beta1.ClusterPolicySnapshot) (int, erro
 	return v, nil
 }
 
+// parseResourceIndexFromLabel returns error when parsing the label which should never return error in production.
+func parseResourceIndexFromLabel(s *fleetv1beta1.ClusterResourceSnapshot) (int, error) {
+	indexLabel := s.Labels[fleetv1beta1.ResourceIndexLabel]
+	v, err := strconv.Atoi(indexLabel)
+	if err != nil {
+		return -1, err
+	}
+	if v < 0 {
```

Comment:
It's just a nit, but the two branches can be merged into one "invalid resource index" error handling I believe.